### DISCUSSION TITLE
Conditionally rendered components unexpected behavior when using the watch() method

### DISCUSSION THREAD

**Discussion by aadlc** [react-hook-form/react-hook-form#4971] [Q&A]
Hello,

In React Hook From V6 when I toggled a component off using the watch method, the component will be unmounted and so will any sibling dependent on the unmounted component:

For example: https://codesandbox.io/s/mui-rfhv6-001-xy39c

- We have three components `<ComponentA />`, `<ComponentB />` and `<ComponentC />`.
- When `<ComponentA />` equals true,  `<ComponentB />` is rendered.
- When `<ComponetB />` equals true,` <ComponentC />` is rendered.
- Finally, if I change `<ComponentA />` to false `<ComponentB />` and `<ComponentC />` will be unmounted.

Working example here:

Now, in React Hook From V7 I tried to accomplish the same behavior but when I change `<ComponentA />` to false,
only `<ComponentB />` is unmounted.

Working example here: https://codesandbox.io/s/mui-rhfv7-001-rpg21

Any ideas why?
Am I missing something?

**Comment by bluebill1049** [comment]
try `shouldUnregister: true`? v7 by default we don't remove input state by unmounting.

**Reply by aadlc** [reply]
Thank you.

Here is what I tried at useForm level and did not work.

```
const { control, watch, handleSubmit } = useForm({
    mode: "all",
    shouldUnregister: true
  });
```

Also, you can look at  the working example https://codesandbox.io/s/mui-rhfv7-001-rpg21?file=/src/App.js

**Reply by bluebill1049** [reply]
<img width="567" alt="Screen Shot 2021-04-27 at 1 27 45 pm" src="https://user-images.githubusercontent.com/10513364/116180534-68ae1100-a75c-11eb-9535-cc7d9a2f6aee.png">

This part didn't work for me.

**Reply by aadlc** [reply]
Hello @bluebill1049 ,
I tried the above but still, get the same result. That said, I figured out why `<CoponentC />` is not toggling off. When `<ComponentB />` is unmounted, its value is not cleared. 

Is there a way of clearing `<ComponentB />` value when unmounted?

Please checkout the working example here  https://codesandbox.io/s/mui-rhfv7-001-rpg21?file=/src/App.js. Now, when you press submit you can see the value of `<ComponentB />` separately from the form's output.
